language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -121,57 +123,63 @@ impl<'a> fmt::Display for Token<'a> {
 
 pub struct Lexer<'input> {
     text: &'input str,
-    consumed: usize,
-    previous_end: usize,
-    token: (usize, Token<'input>, usize),
+    prev_end: usize,
```

Comment:
Nice! I like this implementation much better than what I had ported over from lalrpop.